### ISSUE TITLE
Unnecessary cbindgen failure when dyn token present in function signature

### ISSUE THREAD

**Issue by thombles** [eqrion/cbindgen#385]
I think that a pointer to a boxed trait object should result in a valid function prototype. It appears to be unhappy about the `dyn` keyword. If you leave it out it produces correct/expected output, but doing so now produces a warning on modern `rustc`.

Input file using `dyn`:

```rust
pub trait T {}
struct S;
impl T for S {}

#[no_mangle]
pub extern "C" fn foo() -> *mut Box<dyn T> {
    let t: Box<dyn T> = Box::new(S);
    Box::into_raw(Box::new(t))
}
```

Output:

    >cbindgen --lang C
    ERROR: Cannot use fn struct_basic::foo (Unsupported type: TraitObject(TypeTraitObject { dyn_token: Some(Dyn), bounds: [Trait(TraitBound { paren_token: None, modifier: None, lifetimes: None, path: Path { leading_colon: None, segments: [PathSegment { ident: Ident(T), arguments: None }] } })] })).
    #include <stdarg.h>
    #include <stdbool.h>
    #include <stdint.h>
    #include <stdlib.h>

Input without `dyn` return type:
```rust
pub trait T {}
struct S;
impl T for S {}

#[no_mangle]
pub extern "C" fn foo() -> *mut Box<T> {
    let t: Box<dyn T> = Box::new(S);
    Box::into_raw(Box::new(t))
}
```

Output (correct/expected):

    >cbindgen --lang C
    #include <stdarg.h>
    #include <stdbool.h>
    #include <stdint.h>
    #include <stdlib.h>

    typedef struct Box_T Box_T;

    Box_T *foo(void);

When compiling the version without `dyn`:

    warning: trait objects without an explicit `dyn` are deprecated
    --> src\lib.rs:6:37
    |
    6 | pub extern "C" fn foo() -> *mut Box<T> {
    |                                     ^ help: use `dyn`: `dyn T`
    |
    = note: `#[warn(bare_trait_objects)]` on by default

**Comment by emilio** [comment]
Yeah, this is probably a matter of augmenting the parser to handle this case correctly, should be easy to fix.
